python/cudf/cudf/core/index.py

```diff
@@ -2744,6 +2748,214 @@ def categories(self):
         return self._values.cat().categories
 
 
+def interval_range(
+    start: int = None,
+    end: int = None,
+    periods: int = None,
+    freq: int = None,
```

Comment:
Ideally need to handle strings and dateoffsets here as well. Should likely raise `NotImplementedError` if possible.